crates/bifrost/src/providers/replicated_loglet/tasks/trim.rs

```diff
@@ -155,13 +155,13 @@ impl<'a> TrimTask<'a> {
                 // Let's keep the rest of the trim requests running in the background.
                 debug!(
                     loglet_id = %self.my_params.loglet_id,
-                     trim_point = %trim_point,
+                    trim_point = %trim_point,
                     known_global_tail = %self.known_global_tail,
                     "Loglet has been trimmed"
                 );
                 // continue to run the rest of trim requests in the background
                 inflight_requests.detach_all();
-                return Ok(());
+                return Ok(Some(trim_point));
```

Comment:
It seems as if we are only returning `None` if `trim_point == INVALID`. I think that this implementation does not adhere to your defined contract since if I call this method twice with the same value the second call won't return `None` but `Some(value)`. Btw, I think this is how it should be.